### PR TITLE
bluetooth report battery level to host

### PR DESCRIPTION
**Pull request by myst729**
## Description

This PR enhances the keyboard to report battery level to the host. Tested on my Macbook and iPhone. No keymap config required.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by myst729** [comment]
Looking around the related issues, and found [this one](https://github.com/qmk/qmk_firmware/pull/7019/files) from two years ago. How about now?

File: drivers/bluetooth/adafruit_ble.cpp [hunk 1]
```diff
@@ -673,6 +679,32 @@ void adafruit_ble_send_mouse_move(int8_t x, int8_t y, int8_t scroll, int8_t pan,
 
 uint32_t adafruit_ble_read_battery_voltage(void) { return state.vbat; }
 
+// https://learn.adafruit.com/adafruit-feather-32u4-bluefruit-le/power-management#measuring-battery-2177042-9
```
Comment (myst729) [inline drivers/bluetooth/adafruit_ble.cpp:682]: In consideration of this page, how about to provide it as the default behavior, but also allows user to override?

**Comment by fauxpark** [comment]
Interesting. The last time I tested the GATT battery service it only worked on a then-recent version of Windows 10. Good to know more stuff supports it now.

I put my PR on hold because I wanted to rework the Bluetooth system in QMK into something more platform-agnostic (at the time it was very tied to LUFA and so AVR-only), and also due to the subject of accurate percentage measurement being unresolved. I'm still working on this when I can, and it might be a while until I get to a point where battery reporting can be easily added.

**Comment by myst729** [comment]
Well, the accuracy is a problem. From a pragmatic perspective on the other hand, I think the advantage is not "knowing the accurate quota at anytime", but "extending the battery life". For example, when using a Li-Po battery, I don't care if it's 99% or 68%. But when it reads less than 20%, I know it's time to recharge.

**Comment by myst729** [comment]
As stated above, I didn't notice the prior art from @fauxpark when I sent this PR. In addition, I didn't contribute any new ideas. Close this one and let's see where does #7019 go.

**Pull request closed without merging**